tests/multipleBackend/objectPutPart.js

```diff
@@ -0,0 +1,327 @@
+import assert from 'assert';
+import async from 'async';
+import crypto from 'crypto';
+import { parseString } from 'xml2js';
+
+import { cleanup, DummyRequestLogger, makeAuthInfo } from '../unit/helpers';
+import { ds } from '../../lib/data/in_memory/backend';
+import bucketPut from '../../lib/api/bucketPut';
+import initiateMultipartUpload from '../../lib/api/initiateMultipartUpload';
+import objectPutPart from '../../lib/api/objectPutPart';
+import DummyRequest from '../unit/DummyRequest';
+import { metadata } from '../../lib/metadata/in_memory/metadata';
+import constants from '../../constants';
+
+const log = new DummyRequestLogger();
+const canonicalID = 'accessKey1';
+const authInfo = makeAuthInfo(canonicalID);
+const namespace = 'default';
+const bucketName = 'bucketname';
+const objectName = 'objectName';
+const postBody = Buffer.from('I am a body', 'utf8');
+const mpuBucket = `${constants.mpuBucketPrefix}${bucketName}`;
+const bucketPutRequest = {
+    bucketName,
+    namespace,
+    headers: { host: `${bucketName}.s3.amazonaws.com` },
+    url: '/',
+    post: '',
+};
+let locationConstraint;
+let initiateRequest;
+
+describe('objectPutPart API with multiple backends', () => {
+    beforeEach(() => {
+        cleanup();
+        locationConstraint = 'file';
+        initiateRequest = {
+            bucketName,
+            namespace,
+            objectName,
+            headers: { host: `${bucketName}.s3.amazonaws.com` },
+            url: `/${objectName}?uploads`,
+        };
+    });
+
+    it('should upload a part to file based on mpu location', done => {
+        locationConstraint = 'mem';
```

Comment:
you could create a function to do these tests with the parameters you are changing between tests to avoid the code duplication